src/code/z_actor.c

```diff
@@ -3043,106 +3089,150 @@ Actor* Actor_Delete(ActorContext* actorCtx, Actor* actor, PlayState* play) {
     return newHead;
 }
 
-int func_80032880(PlayState* play, Actor* actor) {
-    s16 sp1E;
-    s16 sp1C;
+int Target_InTargetableScreenRegion(PlayState* play, Actor* actor) {
+    s16 x;
+    s16 y;
 
-    Actor_GetScreenPos(play, actor, &sp1E, &sp1C);
+    Actor_GetScreenPos(play, actor, &x, &y);
 
-    return (sp1E > -20) && (sp1E < 340) && (sp1C > -160) && (sp1C < 400);
+    return (x > -20) && (x < SCREEN_WIDTH + 20) && (y > -160) && (y < SCREEN_HEIGHT + 160);
 }
 
-Actor* D_8015BBE8;
-Actor* D_8015BBEC;
-f32 D_8015BBF0;
-f32 sbgmEnemyDistSq;
-s32 D_8015BBF8;
-s16 D_8015BBFC;
+Actor* sTargetableNearestActor;
+Actor* sTargetablePrioritizedActor;
+f32 sTargetableNearestActorDistSq;
+f32 sBgmEnemyDistSq;
+s32 sTargetablePrioritizedPriority;
+s16 sTargetPlayerRotY;
 
-void func_800328D4(PlayState* play, ActorContext* actorCtx, Player* player, u32 actorCategory) {
-    f32 var;
+/**
+ * Search for targetable actors of the `actorCategory` category.
+ *
+ * Looks for the actor of said category with higher targetPriority and the one that is nearest to player. This actor
+ * must be within the range (relative to player) speicified by its targetMode.
+ *
+ * The actor must be on-screen
+ *
+ * The highest priority actor is stored in `sTargetablePrioritizedActor`, while the nearest actor is stored in
+ * `sTargetableNearestActor`. The higher priority / smaller distance of those actors are stored in
+ * `sTargetablePrioritizedPriority` and `sTargetableNearestActorDistSq`.
+ *
+ * For an actor to be taken in consideration by this function it needs:
+ * - Non-NULL update function
+ * - Not be Player itself.
+ * - It must be targetable.
+ * - Not be the already targeted actor.
+ * - Be withing the range specified by its targetMode.
+ * - It must be on-screen (within a margin)
+ * - Must not be blocked by a surface
+ *
+ * This function also checks for the nearest enemy actor, which allows determining if enemy background music should be
+ * played. This actor is stored in `targetCtx.bgmEnemy` and its distance is stored in `sBgmEnemyDistSq`
+ *
+ * This function is expected to be called with almost every actor category in each cycle. On a new cycle its global
+ * variables must be reset by the caller, otherwise the information of the previous cycle will be retained on this one.
+ */
+void Target_FindTargetableActorForCategory(PlayState* play, ActorContext* actorCtx, Player* player, u32 actorCategory) {
+    f32 distSq;
```

Comment:
So just to make sure, when this system calls an actor "targetable", that means it is either lock-able, or hover-able?

I feel like this is going to be a common place of confusion moving forward, with most people assuming "targetable" means "lockable", cause thats how the term is commonly used. But I guess theres not much we can do about it. I dont have any ideas for a different term.

Reply:
Looking at it again, it's actually very specifically used for "hover-able".

`Target_FindTargetableActorForCategory` is used specifically for `Target_GetTargetActor` and no where else, and `Target_GetTargetActor` is used to get the actor that's set to `targetCtx->arrowHoverActor`. Afterwards, `actor` in that function gets overwritten with something else so it's not used for any other purpose. It's just that `arrowHoverActor` then gets used for other purposes later to allow an actor to be locked-on 

```c
if ((player->unk_664 != NULL) && (player->unk_84B[player->unk_846] == 2)) {
    targetCtx->arrowHoverActor = NULL;
} else {
    Target_GetTargetActor(play, &play->actorCtx, &actor, player);
    targetCtx->arrowHoverActor = actor;
}
```

So maybe:
`Target_GetTargetActor` -> `Target_GetArrowHoverActor`
`Target_FindTargetableActorForCategory` -> `Target_FindArrowHoverActorForCategory`

But then 
```c
Actor* sTargetableNearestActor;
Actor* sTargetablePrioritizedActor;
f32 sTargetableNearestActorDistSq;
s32 sTargetablePrioritizedPriority;
```
Should all change to `ArrowHoverActor`?

Reply:
I was thinking, maybe we shouldnt be so set on this being the "arrow" actor, and more on the fact that its the next lock-able actor?
Does the arrow appear above things that are not lockable?
![image](https://github.com/zeldaret/oot/assets/22045047/fce2b265-7304-4b85-8b67-07b681028911)

Reply:
video helps get the point across better. this "current" and "next" lockable actor interaction is a pretty important concept to get across, and I dont think the current "arrow" naming helps with this. but Ive not investigated all of the code to see if the way it ends up working in practice translates to these functions well

https://github.com/zeldaret/oot/assets/22045047/c2e0335e-f7d9-44c9-9348-4fda9d149911